psi4/src/psi4/libmints/dimension.h

```diff
@@ -37,59 +37,88 @@
 
 namespace psi {
 
+/// @brief Dimension object
+/// \ingroup MINTS
 class PSI_API Dimension {
    private:
     std::string name_;
     std::vector<int> blocks_;
 
    public:
+    /// @brief Constructs an empty Dimension object, with the name initialized to "(empty)"
     Dimension();
-    Dimension(int n, const std::string& name = "");
+
+    /// @brief Constructs a Dimension object with a specified number of block numbers and optionally a name. If no name
+    /// is given it will be a zero-length string.
+    /// @param n : number of blocks
+    /// @param name : (optional) name associated with this Dimension object
+    Dimension(const size_t n, const std::string& name = "");
+
+    /// @brief Constructs a Dimension object from an std::vector<int> object, leaving the name a zero-length string.
+    /// @param other : object to copy the block numbers from
     Dimension(const std::vector<int>& other);
 
-    /// Assignment operator, this one can be very dangerous
+    /// @brief Assignment operator, this one can be very dangerous
+    PSI_DEPRECATED(
+        "The assignment operator for psi::Dimension is being deprecated. Unless someone speaks up, 1.10 may be the "
```

Comment:
```suggestion
        "The assignment operator for psi::Dimension is being deprecated. Unless someone speaks up, v1.9 may be the "
```
1.9 instead? that would mean the fns could be removed before the May 2024 v1.10 release.